### PR TITLE
Reindex on Algolia crawl

### LINKED ISSUE #1349: Crawl Step in Deploy Action Does Not Trigger a Crawl
**Describe the bug**

The crawl step in the deploy action doesn't trigger a Crawl for the IOTA index in Algolia.

**To Reproduce**

Run the deploy action,  and look at the latest crawl time in https://crawler.algolia.com/admin/crawlers/bd7ad9be-f639-4213-8e62-ccdf4d422136/overview. 

**Expected behavior**
The Algolia crawler should be triggered every single deploy so the search indexes are updated.

**Additional context**
It probably has something to do with the Index ID or API Keys.

### PR DESCRIPTION
**Pull request by Dr-Electron**
# Description of change

This PR should force a reindex after the crawler run. I was too lazy to check docs so I just stole from the official [GitHub action ](https://github.com/algolia/algoliasearch-crawler-github-actions/blob/bdf5357c4a4cc49ea728d7d7207acb454113d10a/src/crawler-api-client.ts#L302-L305) :trollface: 
We could also use the GitHub action. But I think in our case it isn't really needed.

## Links to any relevant issues

Fixes #1349.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -42,5 +42,7 @@ jobs:
         run: vercel deploy --prebuilt --prod --token=${{ secrets.VERCEL_TOKEN }}
       # Trigger Algolia crawler
       - name: Crawl
+        uses: algolia/algoliasearch-crawler-github-actions@v1
```
Comment (jlvandenhout) [inline .github/workflows/deploy.yml:45]: Is this really needed when we use the REST API?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -42,5 +42,7 @@ jobs:
...
         run: |
           curl -H "Content-Type: application/json" -X POST --user ${{ secrets.ALGOLIA_USER_ID }}:${{ secrets.ALGOLIA_API_KEY }} "https://crawler.algolia.com/api/1/crawlers/${{ secrets.ALGOLIA_CRAWLER_ID }}/run"
```
Comment (jlvandenhout) [inline .github/workflows/deploy.yml:47]: We can remove this. From the docs I understand it only activates the crawler, it doesn't actually trigger a crawl till the next schedule. The below command should actually trigger a crawl.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (Dr-Electron) [reply .github/workflows/deploy.yml:45]: haha, no. Seems like I just forgot it after trying it out 😅 
```suggestion
```

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -42,5 +42,7 @@ jobs:
...
+          curl -H "Content-Type: application/json" -X POST --user ${{ secrets.ALGOLIA_USER_ID }}:${{ secrets.ALGOLIA_API_KEY }} "https://crawler.algolia.com/api/1/crawlers/${{ secrets.ALGOLIA_CRAWLER_ID }}/reindex"
```
Comment (Dr-Electron) [inline .github/workflows/deploy.yml:48]:
```suggestion
          curl -H "Content-Type: application/json" -X POST --user ${{ secrets.ALGOLIA_USER_ID }}:${{ secrets.ALGOLIA_API_KEY }} "https://crawler.algolia.com/api/1/crawlers/${{ secrets.ALGOLIA_CRAWLER_ID }}/reindex"

```

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (Dr-Electron) [reply .github/workflows/deploy.yml:47]: Yeah, seems correct 🤔 
```suggestion
```

**Review by jlvandenhout — APPROVED** [review]

**Pull request merged**